### PR TITLE
#162171030 list all users functionality

### PR DESCRIPTION
**Pull request by pitaz**
**What does this PR do?**
- List all existing authors

**Description of Task to be completed?**
- modify .eslintrc
- create UsersController.js
- create test case to handle listing out users
- modify models/index.js file
- create user route route/user.js
- modify test file to handle users list functionality
**How should this be manually tested?**
- clone the repository `https://github.com/andela/heimdal-ah-backend.git`
- run `npm run install`
- run `npm run start:dev`
- input `api/v1/users/authors` on postman with the `get` method
**Any background context you want to provide?**
- none
**What are the relevant pivotal tracker stories?**
- [#162171030](https://www.pivotaltracker.com/story/show/162171030)

**Screenshots (if appropriate)**
- NA
**Questions:**
- NA

### CODE REVIEW

File: controllers/UsersController.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import status from '../helpers/StatusResponse';
```
Comment (thebere) [inline controllers/UsersController.js]: Import as Response which is being used in other parts of the application

File: controllers/UsersController.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        include: [profiles, {
+          model: roles,
+          as: 'roles',
+          where: {
+            role: 'author'
+          }
+        }],
+        attributes: { exclude: ['password'] }
+      });
+      if (authorsList) {
```
Comment (thebere) [inline controllers/UsersController.js]: This is not grammatically correct. `authorList` or `authors` not `authorsList`. Also, if it results null or the array is empty, you are not handling for that

File: controllers/UsersController.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        status.success(res, {
+          message: 'List of authors',
+          authors: authorsList
+        });
+      }
+    } catch (error) {
```
Comment (thebere) [inline controllers/UsersController.js:40]: Unused var `error`

File: test/users.js [hunk 4]
```diff
@@ -171,3 +171,16 @@ describe('Test for registering a new user', () => {
     res.body.message.should.equal('This email has been taken');
   });
 });
+
+
+describe('Test for listing all users', () => {
+  it('Should return status 200 on successful retreival Of users', async () => {
+    const response = await chai.request(app)
```
Comment (thebere) [inline test/users.js:178]: `of users` not `Of users`

File: controllers/UsersController.js (hunk 3 above)
Comment (pitaz) [reply controllers/UsersController.js:40]: Thanks Joshua.
 I will resolve this as soon as possible

File: test/users.js (hunk 4 above)
Comment (pitaz) [reply test/users.js:178]: Thanks for pointing this out, it will be resolved shortly.

File: controllers/UsersController.js (hunk 1 above)
Comment (pitaz) [reply controllers/UsersController.js]: Thanks Joshua, I will refactor this immediately.

File: controllers/UsersController.js (hunk 2 above)
Comment (pitaz) [reply controllers/UsersController.js]: Thanks for pointing this out, I will refactor immediately.

**Pull request merged**